### ISSUE TITLE
Ubuntu Jammy ncneofetch issue: ./src/fetch/main.c:811: ncneofetch: Assertion `display_marshal.nextline >= 0' failed.

### ISSUE THREAD

**Issue by satmandu** [dankamongmen/notcurses#2494]
![image](https://user-images.githubusercontent.com/1096701/146787187-b24cd7a0-b02a-4169-973e-4116398a12fc.png)

Please include the following data:
* `export | egrep 'LANG|LC_CTYPE|TERM'`
```
export | egrep 'LANG|LC_CTYPE|TERM'
declare -x LANG="en_US.UTF-8"
declare -x TERM="xterm-256color"
```
* notcurses version (available from `notcurses-demo i`)
`3.0.1`
* terminal name + version
`Alacritty 0.10.0-dev`

**Comment by dankamongmen** [comment]
thanks for the report! i'll get on this today.

**Comment by dankamongmen** [comment]
by the way, is this the main trunk Alacritty, or is this the sixel-patched branch (@ayosec/graphics)? i'm guessing the former?

if you could give me a screenshot of `notcurses-info` output, that would be very helpful.

**Comment by satmandu** [comment]
This is in main/trunk Alacritty.

![image](https://user-images.githubusercontent.com/1096701/146810823-d354b752-b7a1-4126-83b2-4b93e1c6c042.png)

**Comment by dankamongmen** [comment]
wait, this is xterm above, is it not? 

wait....why is your `TERM` set to `xterm-256color`? when i run Alacritty, i get:

```
[schwarzgerat](0) $ export | grep TERM
declare -x COLORTERM="truecolor"
declare -x TERM="alacritty"
```

**Comment by satmandu** [comment]
I have:
```
satadru$export | grep TERM
declare -x COLORTERM="truecolor"
declare -x TERM="xterm-256color"
```

Looking at `.config/alacritty/alacritty.yml` I see:

```
# Configuration for Alacritty, the GPU enhanced terminal emulator.

# Any items in the `env` entry below will be added as
# environment variables. Some entries may override variables
# set by alacritty itself.
#env:
  # TERM variable
  #
  # This value is used to set the `$TERM` environment variable for
  # each instance of Alacritty. If it is not present, alacritty will
  # check the local terminfo database and use `alacritty` if it is
  # available, otherwise `xterm-256color` is used.
  #TERM: xterm-256color
```

**Comment by satmandu** [comment]
I see the same issue using the terminal in macos 12.1/Monterey:

![image](https://user-images.githubusercontent.com/1096701/146813863-8a273d56-ccb4-448d-b6cb-e989704d1efb.png)

**Comment by dankamongmen** [comment]
> #env:
>   # TERM variable
>   #
>   # This value is used to set the `$TERM` environment variable for
>   # each instance of Alacritty. If it is not present, alacritty will
>   # check the local terminfo database and use `alacritty` if it is
>   # available, otherwise `xterm-256color` is used.
>   #TERM: xterm-256color
> ```

interesting. so are you missing an Alacritty terminfo entry? you can check with `infocmp alacritty`. for instance i see:

```
[schwarzgerat](1) $ infocmp alacritty
#	Reconstructed via infocmp from file: /usr/share/terminfo/a/alacritty
alacritty|alacritty terminal emulator,
	am, bce, ccc, hs, mc5i, mir, msgr, npc, xenl,
	colors#0x100, cols#80, it#8, lines#24, pairs#0x10000,
	acsc=``aaffggiijjkkllmmnnooppqqrrssttuuvvwwxxyyzz{{||}}~~,
	bel=^G, blink=\E[5m, bold=\E[1m, cbt=\E[Z, civis=\E[?25l,
	clear=\E[H\E[2J, cnorm=\E[?12l\E[?25h, cr=\r,
	csr=\E[%i%p1%d;%p2%dr, cub=\E[%p1%dD, cub1=^H,
...
```

i don't think this is responsible for the assertion you're seeing, but it'll definitely be suboptimal. for instance, compare my `notcurses-info` in Alacritty to yours:

![2021-12-20-131848_819x638_scrot](https://user-images.githubusercontent.com/143473/146814193-06479d48-9317-48e3-b78e-4142d017b2c8.png)

**Comment by dankamongmen** [comment]
ok, i've reproduced your exact geometry:

```
notcurses 3.0.1 on Alacritty 0.15.1 (Linux 5.15.9nlb)
45 rows (26px) 122 cols (13px) 1170x1586 rgb+256 colors
gcc-11.2.0 (LE)
terminfo 6.3.20211021 zlib 1.2.11 GPM 2.1.0
avformat 59.10.100 avutil 57.11.100 swscale 6.1.101 avcodec 59.14.100
```

let me see if i can trigger this

**Comment by dankamongmen** [comment]
```
[schwarzgerat](0) $ ./ncneofetch
                             ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                             ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                                                      _,met$$$$$gg.
                                                   ,g$$$$$$$$$$$$$$$P.
                                                 ,g$$P"     """Y$$.".
                                                ,$$P'              `$$$.
                                               ',$$P       ,ggs.     `$$b:
                                               `d$$'     ,$P"'   .    $$$
                                                $$P      d$'     ,    $$P
                                                $$:      $$.   -    ,d$$'
                                                $$;      Y$b._   _,d$P'
                                                Y$$.    `.`"Y$$$$P"'
                                                `$$b      "-.__
                                                 `Y$$
                                                  `Y$$.
                                                    `$$b.
                                                      `Y$$b.
                                                         `"Y$b._
                                        (no image file is known for your distro)
                         ─────────────────────────[ dank@schwarzgerat ]──────────────────────────
                         │Linux 5.15.9nlb                          Debian GNU/Linux bookworm/sid│
                         │RAM: 65.06GiB/74.54GiB                                 Processes: 1543│
                         │DM: XFCE                                              Shell: /bin/bash│
                         │RGB TERM: Alacritty 0.15.1                           LANG: en_US.UTF-8│
                         │       AMD Ryzen Threadripper 3970X 32-Core Processor (64 cores)      │
                         ╰──────────────────────────────────────────────────────────────────────╯
[schwarzgerat](0) $
```


argh

**Comment by dankamongmen** [comment]
got it! had to override with `ubuntu`

```
[schwarzgerat](0) $ ./ncneofetch
                             ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                             ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
ncneofetch: /home/dank/src/dankamongmen/notcurses/src/fetch/main.c:812: ncneofetch: Assertion `display_marshal.nextline >= 0' failed.
Aborted (core dumped)
[schwarzgerat](134) $
```

ok great thanks

**Comment by satmandu** [comment]
Could this be a kernel issue? I'm running 5.16-rc6

**Comment by satmandu** [comment]
( I see the same issue connecting from Chromebooks.)

**Comment by dankamongmen** [comment]
nope, it's my bug, i've reproduced it now and will have it fixed within a few hours. thanks for the report! much appreciated.

**Comment by dankamongmen** [comment]
alright, i see both the major bug (we assert() out whenever we're lacking a logo) and the particular bug (we don't have an ubuntu logo for whatever reason). resolving now.

though you said you got the same results on a mac, right? can you see whether your mac has the file

`/System/Library/PrivateFrameworks/LoginUIKit.framework/Versions/A/Frameworks/LoginUICore.framework/Versions/A/Resources/apple@2x.png`

?

**Comment by dankamongmen** [comment]
aliased `ubuntu` to `ubuntu_small`:

```
[schwarzgerat](0) $ ./ncneofetch 2>d
                             ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                             ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                                                                    ./+o+-
                                                            yyyyy- -yyyyyy+
                                                         ://+//////-yyyyyyo
                                                     .++ .:/++++++/-.+sss/`
                                                   .:++o:  /++++++++/:--:/-
                                                  o:+o+:++.`..```.-/oo+++++/
                                                 .:+o:+o/.          `+sssoo+/
                                            .++/+:+oo+o:`             /sssooo.
                                           /+++//+:`oo+o               /::--:.
                                           +/+o+++`o++o               ++////.
                                            .++.o+++oo+:`             /dddhhh.
                                                 .+.o+oo:.          `oddhhhh+
                                                  +.++o+o``-````.:ohdhhhhh+
                                                   `:o+++ `ohhhhhhhhyo++os:
                                                     .o:`.syhhhhhhh/.oo++o`
                                                         /osyyyyyyo++ooo+++/
                                                             ````` +oo+++o:
                                        (no image file is known for your distro)
                         ─────────────────────────[ dank@schwarzgerat ]──────────────────────────
                         │Linux 5.15.9nlb                          Debian GNU/Linux bookworm/sid│
                         │RAM: 64.99GiB/74.54GiB                                 Processes: 1545│
                         │DM: XFCE                                              Shell: /bin/bash│
                         │RGB TERM: Alacritty 0.15.1                           LANG: en_US.UTF-8│
                         │       AMD Ryzen Threadripper 3970X 32-Core Processor (64 cores)      │
                         ╰──────────────────────────────────────────────────────────────────────╯
[schwarzgerat](0) $
```

**Comment by dankamongmen** [comment]
can you verify the existence of `/usr/share/unity/icons/launcher_bfb.png` on your ubuntu machine?

**Comment by dankamongmen** [comment]
all `assert()`s ought have ceased. ubuntu ought now show one logo or another.

**Comment by dankamongmen** [comment]
let me know any further problems you see, but this ought be fixed!

**Comment by satmandu** [comment]
> `/usr/share/unity/icons/launcher_bfb.png`

ls /usr/share/unity/icons/launcher_bfb.png
ls: cannot access '/usr/share/unity/icons/launcher_bfb.png': No such file or directory

**Comment by satmandu** [comment]
Do you have a new debian package to try? I was just using the package in the jammy repos.. I assume from here: https://launchpad.net/ubuntu/+source/notcurses

**Comment by dankamongmen** [comment]
ahhh, no, this fix will be in the 3.0.2 release which is coming real soon now. i'm pretty confident in it, though =].

if you can find an ubuntu logo somewhere on your filesystem, and let me know the path, i can make sure there's a nice bitmap one (not that this will help alacritty anyway).

anyway, i'll cut 3.0.2 before this weekend almost certainly. it usually hits ubuntu a day or two after i upload it to debian unstable.

it's still weird to me that you don't have the alacritty terminfo, though! do you have the `ncurses-term` package installed? you'll want it if you're running alacritty on debian/ubuntu:

```
[schwarzgerat](0) $ apt-file search alacritty | grep ncurses
ncurses-term: /usr/share/terminfo/a/alacritty
ncurses-term: /usr/share/terminfo/a/alacritty+common
ncurses-term: /usr/share/terminfo/a/alacritty-direct
[schwarzgerat](0) $ 
```

though even without that, if you export `COLORTERM=24bit`, you ought get RGB color (which you're lacking currently). the difference will be obvious in e.g. `notcurses-demo` or `notcurses-info`.

**Comment by dankamongmen** [comment]
@satmandu it looks like 3.0.2 has entered Ubuntu Proposed, and ought hit you in a day or two
